kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -239,8 +294,153 @@ unifyOverloadingVsOverloaded
         then return $ Pair overloadingTerm overloadedTerm'
         else throwBottom "different injected ctor"
 
+{- Handles the case
+    overloadingTerm@(overloadingHead(overloadingChildren))
+    vs.
+    inj{S2,injTo}(overloadedVariable)
+  If there exists a (unique) maximum overloadedHead with its result sort S2'
+  within S2 such that
+    inj{S2',injTo}(overloadedHead(freshVars)) =
+        overloadingHead(inj2(freshVars))
+
+  Then it narrows overloadedVariable to inj{S2',S2}(overloadedHead(freshVars))
+  and reduces the initial problem to
+    overloadingTerm@(overloadingHead(overloadingChildren))
+   vs
+    overloadingHead(inj2(freshVars))
+-}
+unifyOverloadingVsOverloadedVariable
+    :: MonadSimplify unifier
+    => InternalVariable variable
+    => Symbol
+    -> TermLike variable
+    -> ElementVariable variable
+    -> Inj ()
+    -> UnifyOverloadingResult unifier variable
+unifyOverloadingVsOverloadedVariable
+    overloadingHead
+    overloadingTerm
+    overloadedVar
+    injProto@Inj { injFrom }
+  = do
+    OverloadSimplifier { isOverloaded, getOverloadedWithinSort }
+            <- Simplifier.askOverloadSimplifier
+    Monad.guard (isOverloaded overloadingHead)
+    case getOverloadedWithinSort injProto overloadingHead injFrom of
+        Right Nothing -> notUnifiableOverloads
+        Right (Just overHead) -> do
+            narrowing@Narrowing { overloadPair = Pair _ overloadedTerm'}
+                <- computeNarrowing
+                    overloadingHead
+                    injProto
+                    freeVars
+                    overloadedVar
+                    overHead
+            return $ WithNarrowing narrowing
+                { overloadPair = Pair overloadingTerm overloadedTerm' }
```

Comment:
Why is it necessary to modify the `overloadPair` here?

Reply:
The problem is that computeNarrowing does not have the entire picture, and can be called in two separate scenarios. Therefore, it returns a Pair having the overloadedTerm' in both sides....

Reply:
I've given two extra arguments to `computeNarrowing` such that now it directly computes the final narrowing, thus removing the need for artificial alterations as the above. thanks!